tree/tree/src/TTree.cxx

```diff
@@ -4703,8 +4704,9 @@ Int_t TTree::Fill()
    // to the case where the tree is in the top level directory.
    if (fDirectory)
       if (TFile *file = fDirectory->GetFile())
-         if ((TDirectory *)file == fDirectory && (file->GetEND() > fgMaxTreeSize))
-            ChangeFile(file);
+         if (static_cast<TDirectory *>(file) == fDirectory && (file->GetEND() > fgMaxTreeSize))
+            if (!(dynamic_cast<ROOT::Experimental::TBufferMergerFile *>(file)))
```

Comment:
```suggestion
            if (!(dynamic_cast<TMemFile*>(file)))
```
is probably as clear (abeit, the non TBufferMergeFile would not reach here because of line 4705 ... but on the other hand users might have other classes derived from TMemFile).

Reply:
i thought we shunned `dynamic_cast` and preferred using `TClass` & company for RTTI

Reply:
@Axel-Naumann ? [For equality TClass is faster, for base class dynamic_cast might be faster]

Reply:
Performance is likely not a concern here. This function is called very rarely...

Reply:
I updated with a cast to `TMemFile`. I'll gather more information on the usage of TClass for RTTI before merging the PR

Reply:
For the record, casting an object of the stated type through a `TObject*` to a `TNamed*`:
```
-----------------------------------------------------------
Benchmark                 Time             CPU   Iterations
-----------------------------------------------------------
BM_stddyn_TNamed       14.5 ns         14.5 ns     48150881
BM_TClass_TNamed       15.4 ns         15.4 ns     45324619
BM_stddyn_TExec        27.9 ns         27.9 ns     25078317
BM_TClass_TExec        36.3 ns         36.3 ns     19301438
BM_stddyn_TBits        27.1 ns         27.1 ns     25844223
BM_TClass_TBits         338 ns          338 ns      2081282
BM_stddyn_2nd          31.3 ns         31.3 ns     22456278
BM_TClass_2nd          48.3 ns         48.3 ns     14494924
```
("2nd" is "a class where TNamed is the second base".) Where do you want me to upload that code?

Reply:
Great, thanks!